### PR TITLE
feat: query and update animals table

### PR DESCRIPTION
**Pull request by taher-software**
Hi,
In this branch, I've brought the following changes to the animals table: :100: 

- Add a column species of type string to the animals table. :+1: 
- Insert new records to the animals table. :+1: 
- Create and undo some transactions. :+1: 
- query the table animals to get some valuable information. :+1: 

So please take a look at my work and give me your feedback before to move forward the next stage.

Cheers,

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -8,4 +8,22 @@ INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts)
 VALUES ('Pikachu', '2021-01-07', 15.04, FALSE, 1);
```
Comment (SajjadAhmad14) [inline data.sql:9]: Please add a new column to the `animals` table called `species` with a string type in `schema.sql`.

File: data.sql [hunk 2]
```diff
@@ -8,4 +8,22 @@ INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts)
...
 
 INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts) 
-VALUES ('Devimon', '2017-05-12', 11, true, 5);
\ No newline at end of file
+VALUES ('Devimon', '2017-05-12', 11, TRUE, 5);
```
Comment (SajjadAhmad14) [inline data.sql:11]: Please add the queries in the `queries.sql` file for following mentioned scenarios:

Please UPDATE the animals table by setting the species column to unspecified, then roll back the change. For
example:
  `BEGIN; -- start transaction
   UPDATE animals
   SET species = 'unspecified'; -- make change
   SELECT species from animals; -- verify that change was made
   ROLLBACK;
   SELECT species from animals; -- verify that change was undone`

Please UPDATE the animals table by setting the species column to digimon for all animals that have a name ending in mon.
please UPDATE the animals table by setting the species column to pokemon for all animals that don't have species already set. For example:
`  BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 
   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
`
Please delete all records in the animals table, then roll back the transaction. For example:
 ` BEGIN;
   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;
   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
`
Please DELETE all animals born after Jan 1st, 2022
please Create a savepoint for the transaction
Please UPDATE all animals' weight to be their weight multiplied by -1.
Then Rollback to the savepoint
Please UPDATE all animals' weights that are negative to be their weight multiplied by -1 and then Commit the transaction. For example 
  ` BEGIN;
    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;
    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP21;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 
    COMMIT;
`
Please create queries to answer following questions:
` How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?
`

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Taher, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 2 above)
Comment (taher-software) [reply data.sql:11]: Hi @SajjadAhmad14 
THank you for your feedback.
I've done all of theses requirements, but I encountered a strange issue that I didn't find all my commit in the repo. I didn't know the reason but I pushed again after make some informal changes.
Sorry again for this copy.

File: queries.sql [hunk 3]
```diff
@@ -14,3 +14,17 @@ SELECT * FROM animals WHERE name != 'Gabumon';
```
Comment (SajjadAhmad14) [inline queries.sql:14]: Please create queries to answer the following questions in transactions:

Please UPDATE the animals table by setting the species column to unspecified, then roll back the change. For
example:
  `BEGIN; -- start transaction
   UPDATE animals
   SET species = 'unspecified'; -- make change
   SELECT species from animals; -- verify that change was made
   ROLLBACK;
   SELECT species from animals; -- verify that change was undone`

Please UPDATE the animals table by setting the species column to digimon for all animals that have a name ending in mon.
please UPDATE the animals table by setting the species column to pokemon for all animals that don't have species already set. For example:
`  BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 
   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
`
Please delete all records in the animals table, then roll back the transaction. For example:
 ` BEGIN;
   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;
   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
`
Please DELETE all animals born after Jan 1st, 2022
please Create a savepoint for the transaction
Please UPDATE all animals' weight to be their weight multiplied by -1.
Then Rollback to the savepoint
Please UPDATE all animals' weights that are negative to be their weight multiplied by -1 and then Commit the transaction. For example 
  ` BEGIN;
    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;
    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP21;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 
    COMMIT;
`

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Taher, 
 ### Thank you for working on changes suggested :ok_hand:
## To Improve 🙏
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 3 above)
Comment (taher-software) [reply queries.sql:14]: Hi @SajjadAhmad14 ,
Thank you for your feedback,
I've done all of these transactions, but in the requirements of the project they are not required to be reported in any files of the repo.[Project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md)

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Taher, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Please leave any questions or comments in the PR if you have any doubts or unclarity!
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**